linked_lists/intersection.py

```diff
@@ -7,8 +7,27 @@ def __init__(self, value):
         self.next = None
 
 
+
 def intersection_node(headA, headB):
     """ Will return the node at which the two lists intersect.
         If the two linked lists have no intersection at all, return None.
     """
-    pass
\ No newline at end of file
+
+
+    cur_A = headA
+    cur_B = headB
+
+    count  = 0
+    while cur_A!=cur_B:
+        if cur_A.next and cur_B.next:
```

Comment:
It looks like some test cases end with an infinite loop.

In test case two, once the shorter list B reaches its tail and `cur_B` is set to the tail node, in the next iteration of the while loop the if statement `if cur_A.next and cur_B.next` is no longer True. Your if statement that might help you exit the loop is nested inside, so you you end up stuck in an infinite loop!